src/transformers/pipelines.py

```diff
@@ -1664,6 +1665,270 @@ def __call__(
+        """
+        # Parse arguments
+        inputs = self._args_parser(*args, **kwargs)
+        inputs = self.tokenizer.batch_encode_plus(inputs, add_special_tokens=False, padding=False).get("input_ids", [])
+        for input in inputs:
+            input.append(self.tokenizer.eos_token_id)
+        return inputs
+
+    def _clean_padding_history(self, generated_tensor) -> List[List[int]]:
+        """
+        Cleans the padding history. Padding may be generated in two places when multiple conversations are provided as
+        an input:
+            - at the end of the concatenated history and new user input, so that all input to the model have the same
+                length
+            - at the end of the generated response, as some responses will be longer than others
+        This method cleans up these padding token so that the history for each conversation is not impacted by the
+        batching process.
+        """
+        outputs = []
+        for sequence in generated_tensor:
+            sequence_tokens = []
+            is_previous_pad = False
+            for token in sequence:
+                if token == self.pad_token_id:
+                    if is_previous_pad:
+                        continue
+                    else:
+                        is_previous_pad = True
+                else:
+                    is_previous_pad = False
+                if self.framework == "pt":
+                    sequence_tokens.append(token.item())
+                else:
+                    sequence_tokens.append(int(token.numpy()))
+
+            outputs.append(sequence_tokens)
+        return outputs
+
+    def _concat_inputs_history(self, inputs: List[List[int]], histories: List[Optional[List[int]]], max_length: int):
+        """
+        Builds an input prepended by the history for this conversation, allowing multi-turn conversation with context
+        """
+        outputs = []
+        for input, history in zip(inputs, histories):
+            if history is not None:
+                input = history + input
+            if len(input) > max_length - self.min_length_for_response:
+                cutoff_eos_index = 0
+                while len(input) - cutoff_eos_index > max_length - self.min_length_for_response:
+                    cutoff_eos_index = input[cutoff_eos_index:].index(self.tokenizer.eos_token_id)
+                    if cutoff_eos_index == 0 or cutoff_eos_index == len(input):
+                        break
+                    else:
+                        input = input[cutoff_eos_index + 1 :]
+            outputs.append(input)
+        max_len = max([len(item) for item in outputs])
+        outputs = [output + [self.pad_token_id] * (max_len - len(output)) for output in outputs]
+        outputs = self.tokenizer.batch_encode_plus(
+            outputs, add_special_tokens=False, is_pretokenized=True, return_tensors=self.framework, padding=False
+        )
```

Comment:
If I understand correctly outputs is already list of list of ints before this line so why do we pass it to `self.tokenizer`? Is it because it's the simplest way to tensorize to pytorch vs tf? (if it is, the intent is not super explicit)

Reply:
Yes - this was the intent. I didn't find a utility class in the library that would conveniently convert a list of integers to a Torch or Tensorflow tensor the way this class did. Is there a better way?

Reply:
Can you create a helper to do this in this file?

Reply:
I actually use the `BatchEncoding` directly from the generated `input_ids`. Cleans up quite a bit and better represents the intent, thank you for the suggestion